test/CountedPtrBenchmarks.h

```diff
@@ -0,0 +1,282 @@
+        TestLog << OutputI << " - Creating, Dereferencing and Copying a CountedPtr " << TestCount
+                << " times with internal counting took: " << CountPtrCopyInternalTime.count() << " Microseconds."
+                << "\n";
+
+        /// External Reference Counter
+        Begin = ClockType::now();
+        for( Whole Count = 0 ; Count < TestCount ; Count++ )
+        {
+            CountedPtr<FooExternal> PtrE( new FooExternal(&destructFlag, 8) );
+            CountedPtr<FooExternal> PtrE2(PtrE);
+            OutputE = PtrE2->Value;
+        }
+        End = ClockType::now();
+        DurationType CountPtrCopyExternalTime = End - Begin;
+        TestLog << OutputE << " - Creating, Dereferencing and Copying a CountedPtr " << TestCount
+                << " times with external counting took: " << CountPtrCopyExternalTime.count() << " Microseconds."
+                << "\n";
+
+
+        /// shared_ptr
+        Begin = ClockType::now();
+        for( Whole Count = 0 ; Count < TestCount ; Count++ )
+        {
+            std::shared_ptr<FooExternal> PtrS( new FooExternal(&destructFlag, 9) );
+            std::shared_ptr<FooExternal> PtrS2(PtrS);
+            OutputS = PtrS2->Value;
+        }
+        End = ClockType::now();
+        DurationType SharedPtrCopyTime = End - Begin;
+        TestLog << OutputS << " - Creating, Dereferencing and Copying a shared_ptr " << TestCount
+                << " times with external counting took: " << SharedPtrCopyTime.count() << " Microseconds."
+                << "\n";
+
+        /// make_shared
+        Begin = ClockType::now();
+        for( Whole Count = 0; Count < TestCount ; Count++)
+        {
+            std::shared_ptr<FooExternal> PtrS = std::make_shared<FooExternal>(&destructFlag, 10);
+            std::shared_ptr<FooExternal> PtrS2(PtrS);
+            OutputS = PtrS2->Value;
+        }
+        End = ClockType::now();
+        DurationType MakeSharedCopyTime = End - Begin;
+        TestLog << OutputS << " - Creating, Dereferencing and Copying a shared_ptr from make_shared "
+                << TestCount << " times with external counting took: " << MakeSharedCopyTime.count() << " Microseconds."
+                << "\n\n";
+
+        /////////////////////////////////////
+        // Comparison of Benchmark Results
+
+        TestLog << "Checking that raw pointers are fastest for sanity: "
+                << ( RawPtrTime < CountPtrInternalTime && RawPtrTime < CountPtrExternalTime
+                    && RawPtrTime<SharedPtrTime && RawPtrTime < MakeSharedTime )
+                << " and "
+                << ( RawPtrCopyTime < CountPtrCopyInternalTime && RawPtrCopyTime < CountPtrCopyExternalTime
+                    && RawPtrCopyTime < SharedPtrCopyTime && RawPtrCopyTime < MakeSharedCopyTime )
+                << ".\n";
+
+        //TEST_WARN( "SanityRawVsCountedPtr", RawPtrTime < CountPtrInternalTime && RawPtrTime < CountPtrExternalTime );
```

Comment:
If this commented out code is extra, can it be removed?

Reply:
After Talking with @MakoEnergy it sounds like you were having some issues with the timings? Anything I can help with? 

We could leave the code commented out until I can circle back around to it unless you really want to dive in.

Reply:
They currently cause some unstable results on certain environments.  Removing them wouldn't be awful, since the TestLog output has the information anyway, but I'm leaning toward leaving them in and readdressing it later.